### PR TITLE
Add deb and rpm package upgrade validation

### PR DESCRIPTION
**Pull request by guicaulada**

### CODE REVIEW

**Review by kminehart — CHANGES_REQUESTED** [review]
I think the approach makes sense!

File: pipelines/package_validate.go [hunk 1]
```diff
@@ -239,3 +252,72 @@ func validateLicense(ctx context.Context, service *dagger.Container, licensePath
...
+	}
+
+	if strings.HasSuffix(firstName, ".rpm") {
+		return nil
+	}
+
+	return fmt.Errorf("invalid upgrade package extension")
+}
+
+func validateDebUpgrade(ctx context.Context, d *dagger.Client, packages []*dagger.File, names []string) error {
```
[the remark this answers is not in the document]
Comment (guicaulada) [reply pipelines/package_validate.go:283]: Yes, that's exactly what it does, I will add some comments to explain it better.

File: pipelines/package_names.go [hunk 2]
```diff
@@ -78,7 +78,7 @@ func TarOptsFromFileName(filename string) TarFileOpts {
 		// arm-7 should become arm/v7
 		arch = strings.Join([]string{archv[0], archv[1]}, "/")
 	}
-	edition := ""
+	edition := "oss"
```
Comment (kminehart) [inline pipelines/package_names.go:81]: On second thought I'm not ok with this change. The edition is not "oss" the edition is nothing. I think this is reflected in the test that's failing

**Review by kminehart — CHANGES_REQUESTED** [review]

File: pipelines/package_names.go (hunk 2 above)
Comment (kminehart) [reply pipelines/package_names.go:81]: I really want us to get away from calling "Grafana" "Grafana OSS". It's just "Grafana".

Comment (guicaulada) [reply pipelines/package_names.go]: Done on d68c40685d00989c4f83364949af6ae6d0c084fe

File: pipelines/package_validate.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (guicaulada) [reply pipelines/package_validate.go:288]: Done on https://github.com/grafana/grafana-build/pull/94/commits/4531b9dd3e8e048a34049e199ee5e871b53fdbe7

File: pipelines/package_validate.go [hunk 3]
```diff
@@ -239,15 +239,19 @@ func validateTarball(ctx context.Context, d *dagger.Client, pkg *dagger.File, sr
...
+		if !strings.Contains(license, "Grafana Enterprise") {
+			return fmt.Errorf("license in package does not match edition in package name")
 		}
 	}
 
 	if taropts.Edition == "" {
-		if err != nil || !strings.Contains(license, "GNU AFFERO GENERAL PUBLIC LICENSE") {
-			return fmt.Errorf("failed to validate open-source license")
+		if !strings.Contains(license, "GNU AFFERO GENERAL PUBLIC LICENSE") {
+			return fmt.Errorf("license in package does not match edition in package name")
```
Comment (kminehart) [inline pipelines/package_validate.go:254]: in these contexts we know what to expect; we should be clearer about what we expect vs what was given. "license in package is not the Grafana Enterprise license agreement", for example.

**Review by kminehart — APPROVED** [review]

**Pull request merged**